### DISCUSSION TITLE
Bug Reports

### DISCUSSION THREAD

**Discussion by Gazook89** [naturalcrit/homebrewery#2366] [Ideas]
I am annoyed\* with the number of Issues opened for saving issues.  For one thing, for many homebrewery users, it means having to create a Github account just to post the Issue.  Likely many of those people have never heard of github before then and are unlikely to come back to check status of their Issue since the Saving issue likely resolves itself fairly quickly (I think).  Also, there has got to be a way to just reduce the creation of new Issues for a recurring problem.  Obviously just being sure that no save error occurs is preferable, but barring that it might be good to consider:

1. A modal in HB that catches the error and gives a possible resolution right then and there (i honestly thought this was already done?).   Or, offers a link to a new `/troubleshooting/` page with options *and then* offers a link to create a github issue.
2. Github Issue Forms as detailed [here](https://beyondco.de/blog/github-issue-forms), which can provide a few solutions at the top, followed by the actual report field.   Fields like "Please describe what you attempted to do which failed" could be made required.
3. Figure out a Github Action that automatically applies the "saving" tag and replies to the original Issue with suggestions.  It could be paired with the above, such that the "saving" tag is applied based on a "Is this issue related to saving" checkbox.  Further, I believe github actions can also close Issues if no update has been made in a certain amount of time.

*\*Not actually really annoyed, but it is an annoyance*

**Comment by ericscheid** [comment]
Much of the advice has been of the form "oops, server was busy, wait a moment and try again".

That could be incorporated into a modal response, even offering a save button there.

Also, if the save was triggered by just the idle-polling, maybe have a number of periodic fails be permitted before throwing the error message?

**Comment by calculuschild** [comment]
I sympathize with the annoyance. I also do not like the constant save reports. However, we need to make sure any solution we apply does not filter out actual issues that we need to solve. For instance, the current flood of saving issues is all related to the same problem, detailed in #2322 , which @jeddai is working on a fix for. However, because it is one of those "temporary" issues that goes away if the user waits a few seconds, directing users to a `/troubleshooting/` page that tells them to "wait and then try again", would mean we never hear about the problem. It's a tricky situation to balance too many reports against no reports at all. As for your solutions:

1) We indeed do have multiple message boxes with instructions for specific issues (Google Credentials expired, file is in Google Trash, not Logged in but trying transfer brew, etc. This works because they are "expected" errors and we know it is not our fault due to a bug, but a user error that can be solved. However every other error lands in a generic "Looks like there was a problem saving. Report the issue here" popup. These issues we definitely want to get reports for else we won't know there is a problem. And if we get 100 identical reports, it's annoying, but we also know it's a legitimate issue and not just a temporary glitch. Don't want to lose that valuable data if a `/troubleshooting/` page temporarily solves the issue but it means we never get any report.

One issue we might be able to add a specific modal for is when the server is offline. Every so often somebody tries to save at the exact moment the server reboots in the middle of the night (we have a daily reboot), or when an issue from our host causes our server to be inaccessible for a few minutes. These are usually 503 errors. On the other hand, we have had problems before where people were causing the server to crash by having too much garbage in their brews (giant base64 images, or ugly strings that regex choked on), and then we would get a string of 503 reports from users. If we blocked all of those reports, we might not have been able to pinpoint the signal that was causing the crashes. I don't know... Perhaps something like "Looks like our servers are temporarily down. Wait a minute and try again. If this keeps happening, report the issue here"

2) I like the look of the issue forms. We could definitely make the "report" button open a specific form for Save issues. But as you say, many users are not familiar with Github. We wouldn't want to add too many required fields that people can't figure out how to even file the report at all.

3) This sounds reasonable. This way we still get the report, but don't have to always process the issue by hand if it happens to be a duplicate, plus the user might get a more immediate solution in case they never think to check back on their Github account.

**Comment by Gazook89** [comment]
I did #2367 as a PR just now because I could and I was tired of a different branch i'm working on.

**Comment by 5e-Cleric** [comment]
While most of this is resolved, we still get tons of 503 issues, as well as some 500 issues and 403:

- 503 - Server busy, we should try to fix this issue, i suggest tuning down the time for automatic saving from... 3s? to 10s. This should lighten the load of the server. (Example issue: https://github.com/naturalcrit/homebrewery/issues/3601)

- 500 - Unknown issue, example (https://github.com/naturalcrit/homebrewery/issues/3600)
